src/core/render/compiler.js

```diff
@@ -225,7 +225,7 @@ export class Compiler {
       nextToc.slug = url;
       _self.toc.push(nextToc);
 
-      return `<h${level} id="${slug}"><a href="${url}" data-id="${slug}" class="anchor"><span>${str}</span></a></h${level}>`;
+      return `<h${level} id="${slug}" tabindex="-1"><a href="${url}" data-id="${slug}" class="anchor"><span>${str}</span></a></h${level}>`;
```

Comment:
A comment to explain the tabindex would be nice.

Reply:
Done.

> // Note: tabindex="-1" allows programmatically focusing on heading
> // elements after navigation. This is preferred over focusing on the link
> // within the heading because it matches the focus behavior of screen
> // readers when navigating content.

Specifically, focusing on the heading instead of the link will cause screen readers to behave the same way they would when navigating page content. This is because screen readers will "focus" on heading elements even though they are not typically elements that can receive focus. Note that this only occurs when navigating content using screen reader navigation keys, not the Tab key (e.g., CapsLock + Up/Down). The result is that when focused on a heading, screen readers will describe the heading before describing the heading contents. When focused on a link within a heading, screen readers will describe the link and its content before the heading level.

- Focus on heading: "Heading level [level]. Visited. Link. [heading text]."
- Focus on link within heading: "Visited. Link. [heading text]. You are currently on a heading level [level]."

From a user perspective, it also makes more sense to click a link and land on a heading since this is standard behavior for anchor links. It is much less common and potentially confusing to click on a link and land on another link.